src/Agent.Listener/Configuration/FeatureFlagProvider.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Agent.Sdk;
+using Microsoft.VisualStudio.Services.Agent;
+using Microsoft.VisualStudio.Services.Agent.Listener.Configuration;
+using Microsoft.VisualStudio.Services.Agent.Util;
+using Microsoft.VisualStudio.Services.Common;
+using Microsoft.VisualStudio.Services.FeatureAvailability;
+using Microsoft.VisualStudio.Services.FeatureAvailability.WebApi;
+using System;
+using System.Threading.Tasks;
+
+namespace Agent.Listener.Configuration
+{
+    [ServiceLocator(Default = typeof(FeatureFlagProvider))]
+    public interface IFeatureFlagProvider : IAgentService
+    {
+        /// <summary>
+        /// Gets the status of a feature flag from the specified service endpoint.
+        /// If request fails, the feature flag is assumed to be off.
+        /// </summary>
+        /// <param name="context">Agent host contexts</param>
+        /// <param name="featureFlagName">The name of the feature flag to get the status of.</param>
+        /// <param name="traceWriter">Trace writer for output</param>
+        /// <returns>The status of the feature flag.</returns>
+        /// <exception cref="VssUnauthorizedException">Thrown if token is not suitable for retriving feature flag status</exception>
+        /// <exception cref="InvalidOperationException">Thrown if agent is not configured</exception>
```

Comment:
```suggestion
```

Reply:
@merlynomsft this exception might be thrown in case if agent not configured (it is pretty rare cases, but for convenience I think it should be here)

Reply:
either I highlighted the wrong line or it shifted due to code changes.  I meant to comment on the comment containing   VssServiceException , which I think was removed